mne/beamformer/_ap.py

```diff
@@ -0,0 +1,743 @@
+        forward = convert_forward_solution(forward, surf_ori=False)
+    is_free_ori, info, _, _, gain, whitener, _, _ = _prepare_beamformer_input(
+        info, forward, noise_cov=noise_cov, rank=None)
+    forward = pick_channels_forward(forward, info['ch_names'], ordered=True)
+    del info
+
+    # whiten the data (leadfield already whitened)
+    wh_data = np.dot(whitener, data)
+    del data
+
+    if is_free_ori:
+        idx, oris, poss, gain_active, gain_dip, sol, dip_ind = \
+            free_ori_ap(wh_data, gain, nsources, forward,
+                        max_iter=max_iter)
+        X = sol[:, np.newaxis] * oris[:, :, np.newaxis]
+        X.shape = (-1, len(times))
+    else:
+        idx, oris, poss, gain_active, gain_dip, sol = \
+            fixed_ori_ap(wh_data, gain, nsources, forward,
+                         max_iter=max_iter)
+        X = sol
+        dip_ind = idx
+
+    gain_active = whitener @ gain_active.reshape(gain.shape[0], -1)
+    explained_data = gain_dip @ sol
+    m_estimate = whitener @ explained_data
+    var_exp = _log_exp_var(wh_data, m_estimate)
+    tstep = np.median(np.diff(times)) if len(times) > 1 else 1.
+    dipoles = _make_dipoles_sparse(
+        X, idx, forward, times[0], tstep, wh_data,
+        gain_active, active_is_idx=True)
+    for dipole, ori in zip(dipoles, oris):
+        signs = np.sign((dipole.ori * ori).sum(-1, keepdims=True))
+        dipole.ori *= signs
+        dipole.amplitude *= signs[:, 0]
+    logger.info('[done]')
+
+    return dipoles, explained_data, var_exp, dip_ind, oris, poss
+
+
+def _residual_packing(evoked, picks, explained_data_mat, info):
+    """Pack the residual data into an mne.Evoked object."""
+    residual = evoked.copy()
+    selection = [info['ch_names'][pick] for pick in picks]
+
+    residual = pick_channels_evoked(residual,
+                                    include=selection)
+    residual.data -= explained_data_mat
+    active_projs = [proj for proj in residual.info['projs'] if proj['active']]
+    for proj in active_projs:
+        proj['active'] = False
+    residual.add_proj(active_projs, remove_existing=True)
+    residual.apply_proj()
+    return residual
+
+
+def _explained_data_packing(evoked, picks, explained_data_mat, info):
+    """Pack the explained data into an mne.Evoked object."""
+    info = evoked.info
+    explained_data = evoked.copy()
+    selection = [info['ch_names'][pick] for pick in picks]
+
+    explained_data = pick_channels_evoked(explained_data,
+                                          include=selection)
+    explained_data.data = explained_data_mat
+    active_projs = [proj for proj in
+                    explained_data.info['projs'] if proj['active']]
+    for proj in active_projs:
+        proj['active'] = False
+    explained_data.add_proj(active_projs, remove_existing=True)
+    explained_data.apply_proj()
+    return explained_data
+
+
+@verbose
+def ap(evoked, forward, nsources, noise_cov=None, max_iter=6,
```

Comment:
```suggestion
def alternating_projection(evoked, forward, nsources, noise_cov=None, max_iter=6,
```
`ap` is too cryptic. We tend to avoid acronyms if we can

Reply:
done